code/ui/swe6813-ui/src/pages/AuthPage/AuthPage.jsx

```diff
@@ -32,80 +32,58 @@ const Box = styled.div`
     background-color: #FFF;
     box-shadow: 0px 4px 4px 0px #444;
 `;
-const Header = styled.h1`
-    font-size: 32px;
-    padding-bottom: 0px;
-`;
-const LoginButton = styled.button`
-    width: 100%;
-    margin: 0;
-`;
 
 function AuthPage () {
-    // TODO: actually check for a real user session
-    const [user, setUser] = useState(null);
-
-    const [activeBox, setActiveBox] = useState('LOGIN')
+    const [activeBox, setActiveBox] = useState('LOGIN');
 
     const navigate = useNavigate();
 
-    const loginUser = useCallback(() => {
-        // TODO: actually login the user before navigating to root app
-        navigate("/");
-
-    }, [navigate]);
-
-    const registerUser = useCallback(() => {
-        // TODO: actually register the user
-    }, [])
-
     const changeBox = useCallback((box) => {
         setActiveBox(box)
     }, [setActiveBox])
 
-    
+    useEffect(() => {
+        const token = localStorage.getItem('token');
+        if (!token) {
+            return
+        }
+        
+        checkToken(token)
+            .then((isValid) => {
+                if (!isValid) {
+                    localStorage.removeItem('token')
+                    return
+                }
+
+                navigate('/')
+            })
+            .catch(console.log)
+    }, [])
```

Comment:
same function as in Root.jsx, right? Should we put it somewhere it can be imported in multiple places?